src/tests/test_command.py

```diff
@@ -5,12 +5,41 @@
 
 from sql.command import SQLCommand
 
+from conftest import runsql
+
 
 @pytest.fixture
 def sql_magic(ip):
     return ip.magics_manager.lsmagic()["line"]["sql"].__self__
 
 
+@pytest.mark.parametrize(
+    "line, query_result",
+    [
+        ("SELECT instr(' ', ' ')", [1]),
+        ("SELECT instr(' a ', ' ')", [1]),
+        ("SELECT instr('aaa', ' ')", [0]),
+        ("SELECT instr('aaa', 'a')", [1]),
+        ("SELECT COUNT(*) FROM test", [2]),
+        ("SELECT * FROM test where n == 1", [(1, 'foo')]),
+        ("SELECT * FROM test", [(1, 'foo'), (2, 'bar')])
```

Comment:
can you add some arguments after and before the query? I think that's what will break this

Reply:
Yes, you are right.
running the following command will return an empty result. ([])
```
%sql --save result SELECT instr(' ', ' ')
``` 

since the cell expects the following structure

```
%%sql --save result
SELECT instr(' ', ' ')
```

If I run this command without this fix it will throw `ValueError: No closing quotation` which I'm not sure what is better.

I can catch this exception and suggest the user use `%%sql` instead.
Another quick option that might be a base for the future parser. We can hold the list of possible arguments and build the cell based on this list and the given line. In that case, we'll need to maintain this list of arguments.

Reply:
Right, the second approach is a kind of heuristics to capture %s of the cases (let's say 70%). 
I think the value error is a bug in its own right?

Reply:
This is proving too complex for an edge case that only a few people will have. I suggest closing this.

Reply:
I agree